### ISSUE TITLE
Add registered users group to dropdown list of groups with whom a collection can be shared

### ISSUE THREAD

**Issue by hannahfrost** [samvera-labs/hyrax#1150]
### Descriptive summary

On the New Collection page, the user should have the option to share a collection with all registered users.

### Rationale

Makes it easy for a user to share a collection with everyone.

### Expected behavior

When I am creating a new collection, and I am ready to select with whom to share it, I need the ability to share it with the group of all users who have repository accounts.

### Actual behavior

I can only share my collection with the admin group, which has very little utility to me as a collection creator who wants to share it more broadly.

![screenshot 2017-06-06 11 26 22](https://user-images.githubusercontent.com/1801076/26845551-a75bb52c-4aab-11e7-8a73-589b1f1ada96.png)

**Comment by mjgiarlo** [comment]
@hannahfrost Do you see this being used to share 'read' access with all registered users, or 'edit' access, or both?

**Comment by hannahfrost** [comment]
@mjgiarlo I had pondered this initially and came to the conclusion that both options should be available. I suspect that 'read' access will be used much more frequently than 'edit', but I can imagine use cases for both. Thanks for checking.

**Comment by mjgiarlo** [comment]
@hannahfrost I ask because granting all registered users "read" is essentially the same as assigning it the 'institution' visibility. But since 'edit' is a requirement, this is new work to be done.

**Comment by atz** [comment]
@mjgiarlo IMO, that is an argument *against* allowing a different representation of "Institution" visibility.  I.E., we should ONLY provide sharing "edit" access to registered users as a group.  Otherwise, for read, we should direct users to change the visibility.

**Comment by hannahfrost** [comment]
@mjgiarlo Right, I realized this while I was riding my bike home from work (seriously). And I think I agree with @atz. Let's talk this over sometime soon, no big rush. The way we are elevating collections now for discovery purposes, it would be helpful to work through how the collection sharing functionality fits.

**Comment by rjkati** [comment]
In Hyrax 3.0.2 the "add group" box for collections is a search box, rather than a drop-down as initially reported. It is possible to add all repository users to a collection by searching for the "registered" group and then adding to the appropriate role. 

![user_collection](https://user-images.githubusercontent.com/32644254/122809839-a0d55a00-d29c-11eb-8532-0137c11f056c.PNG)


While this is technically resolved, the solution is not intuitive to users and may need additional work. @jlhardes

**Comment by jlhardes** [comment]
I think this connects to #227 and possibly the proposed changes from @no-reply in #4686.
